hudi-common/src/main/java/org/apache/hudi/metadata/HoodieTableMetadataUtil.java

```diff
@@ -1361,7 +1363,13 @@ public static Set<String> getValidInstantTimestamps(HoodieTableMetaClient dataMe
         });
 
     // SOLO_COMMIT_TIMESTAMP is used during bootstrap so it is a valid timestamp
-    validInstantTimestamps.add(createIndexInitTimestamp(SOLO_COMMIT_TIMESTAMP, PARTITION_INITIALIZATION_TIME_SUFFIX));
+    List<String> metadataInitializationTimestamps = metadataMetaClient.getActiveTimeline()
```

Comment:
I am also considering if this will give us any benefit. 
validInstantTimestamps are used within LogRecordReader to ignore log blocks which was written using commits which are not completed yet. 

Lets consider diff cases: 
1. for an existing table, we may never use SOLO commit time only since there will def be a latest completed commit from data table that we will use. 
2. For a new table. the base commit time to initialize MDT will be chosen as SOLO COMMIT TIME + suffix (one for each partition being initialized). and bulk insert will kick in. So, the base instant time will have prefix of SOLO COMMIT TIME . but any new log files will be added using new delta commits which will have diff commit times. so, I don't see a necessity to add SOLO COMMIT TIME  to list of valid instant times only.

let me know if I am missing any flow. Just tryin to avoid going through entire active timeline of MDT to filter for SOLO COMMIT TIME if its never going to be used.

Reply:
In other words, 
if valid instant times are used only to filter log blocks, and if there won't be any flow where we might write log blocks with instant times w/ SOLO COMMIT TIME as prefix, we don't need to make this change.

Reply:
> but any new log files will be added using new delta commits which will have diff commit times. 

I agree, the `SOLO_COMMIT_TIMESTAMP` should be kept for backward compatibility BTW.

Reply:
@danny0405 @nsivabalan Agree with your points but should a util method be aware of different cases? Let's say tomorrow for a new MDT partition, the semantics change and it writes a data block with the initializing commit itself, then the author/reviewer needs to come back to util method and fix it. This is going to be harder to maintain. IMO, the better way to handle such cases is by keeping the util method dumb, and do any case handling at the call site or have assertions for invariants such as data block can never have initializing commit time. Wdyt?

Btw, the change is backward compatible as it checks for `startsWith(SOLO_COMMIT_TIMESTAMP)`. Also, there is no reloading of the timeline but to avoid going through active timeline again, i can merge the filter with the stream operation at lines 1349-1350.

Reply:
Valid instants are mainly needed for reading the log files. 
Whereas, first bootstrap commit on the partition, will create one set of dummy delete log blocks and a base file. So, all the data written into partition on the boostrap commit will be present in the base file. Even if dont include these SOLO_COMMIT_TIMESTAMP we should be ok I guess.
But it will be good to include them to keep it consistent.

Reply:
> Even if dont include these SOLO_COMMIT_TIMESTAMP we should be ok I guess.

Yes, that's why I'm saying `SOLO_COMMIT_TIMESTAMP` is for compatibility.